src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -265,6 +265,56 @@ public void RequestAnalysis_Should_NotThrow_When_AnalysisFails()
             act.Should().NotThrow();
         }
 
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow(new string[] { })]
+        public void FilterIssueTrackersByPath_NullOrEmptyPaths_AllTrackersReturned(string[] filePaths)
+        {
+            var tracker1 = CreateMockedIssueTracker("any");
+            var tracker2 = CreateMockedIssueTracker("any2");
+
+            var actual = TaggerProvider.FilterIssuesTrackersByPath(new IIssueTracker[] { tracker1, tracker2}, filePaths);
+            actual.Should().BeEquivalentTo(tracker1, tracker2);
+        }
+
+        [TestMethod]
+        public void FilterIssueTrackersByPath_WithPaths_ExpectedTrackersReturned()
+        {
+            var file1Tracker = CreateMockedIssueTracker("file1.txt");
+            var file2Tracker = CreateMockedIssueTracker("c:\\aaa\\file2.cpp");
+            var file3Tracker = CreateMockedIssueTracker("d:\\bbb\\file3.xxx");
+
+            // 1. No matches
+            filter("no matches", "file1.wrongextension")
```

Comment:
I believe that it's better for the test's readability and "health" to have each use-case tested in a separate test method. This way, when a tests fails, it's way easier to understand what happened. It's also a good way to document the intention of the test, rather than rely on comments.
So for example, I find these to be more self-explanatory: 

FilterIssueTrackersByPath_PathNotNull_NoMatchingTrackers_EmptyListReturned
FilterIssueTrackersByPath_PathNotNull_OneMatchingTracker_OneTrackerReturned
FilterIssueTrackersByPath_PathNotNull_TwoMatchingTrackers_TwoTrackersReturned

Reply:
I've changed the tests. As discussed offline, I find the new version less readable. I think part of the problem for me is that I find long test names very clumsy, and having individual tests for individual conditions makes it harder to tell whether the whole set of necessary conditions for a method have been tested or not.